compiler/hash-ast/src/parse.rs

```diff
@@ -178,22 +197,49 @@ where
                     );
                     let entry_resolver = ParModuleResolver::new(ctx, entry_module_ctx, scope);
 
+                    // @@Cleanup: we need to insert the contents of interactive into the module builder...
+                    //            At the moment, this isn't the cleanest way of going about the problem, we're
+                    //            overwriting the previous content of the interactive session with the current
+                    //            input. This is incorrect because we want to preserve the lines of interactive
+                    //            that are considered to be valid (this is a much deeper problem actually.)
+                    //
+                    //            So, in @@Future, we shouldn't even insert the contents here since it needs
+                    //            to pass the typechecking stage and then be inserted into the line, I guess typechecking
+                    //            can *evict* the current source line if it is invalid?
+                    //
+                    //            @@Copying: we're also having to copy the source because we get a &str instead of a String.
```

Comment:
I wonder if EntryPoint::Interactive should contain a String instead of a &str.

Reply:
I tried changing this before but I ran into some problems, could be worth taking another look.